### DISCUSSION TITLE
Suggest self-hosted applications for Swiftwave One-Click App Store

### DISCUSSION THREAD

**Discussion by tanmoysrt** [swiftwave-org/swiftwave#568] [Ideas]
Swiftwave has an integrated app store that allows you to install any self-hosted app on the server with a single click (Inspired by the `PHP Softaculous store` of Cpanel).

App Store metadata repository: https://github.com/swiftwave-org/app-store.

For testing purposes, we only have one app in the app store. We'd like to add more applications to the app store to make it easier to deploy self-hosted apps.

Please suggest applications that we can add to the Swiftwave App Store that will benefit end users.

| Sr | Tool | Reference | Status | 
| --- | ------ | ------------- | :--------: |
| 1 | Docker Registry | https://hub.docker.com/_/registry | ✅ |
| 2 | Gitea | https://github.com/go-gitea/gitea | ✅ |
| 3 | Filebrowser | https://github.com/filebrowser/filebrowser | ✅ |
| 4 | Ghost | https://github.com/TryGhost/Ghost | ✅ |
| 5 | Minio | https://github.com/minio/minio | 🔨 |
| 6 | WordPress | https://github.com/bitnami/containers/tree/main/bitnami/wordpress#how-to-use-this-image | ✅ |
| 7 | pihole | https://github.com/pi-hole/pi-hole | 🔴 |
| 8 | OhMyForm | https://github.com/ohmyform/ohmyform | 🔨 |
| 9 | Listmonk | https://github.com/knadh/listmonk | 🔨 |
| 10 | Grafana | https://grafana.com/docs/grafana/latest/setup-grafana/installation/docker/ | ✅ |
| 11 | MongoDB | https://hub.docker.com/r/bitnami/mongodb | ✅ |
| 12 | PostgreSQL | https://hub.docker.com/r/bitnami/postgresql | ✅ |
| 13 | MariaDB | https://hub.docker.com/r/bitnami/mariadb | ✅ |
| 14 | MySQL | https://hub.docker.com/r/bitnami/mysql | ✅ |
| 15 | Redis | https://hub.docker.com/r/bitnami/redis | ✅ |
| 16 | PhpMyAdmin | https://hub.docker.com/_/phpmyadmin | ✅ |
| 17 | Pgadmin4 | https://www.pgadmin.org/docs/pgadmin4/latest/container_deployment.html | ✅ |
| 18 | ActivePieces | https://www.activepieces.com/docs/install/options/docker | ❌ |
| 19 | NextCloud | https://github.com/nextcloud/docker | ✅ |
| 20 | Supabase | https://supabase.com/docs/guides/self-hosting/docker | ✅ |
| 21 | Code Server | https://hub.docker.com/r/linuxserver/code-server | ❌ |
| 22 | ClassicPress | https://github.com/ClassicPress/ClassicPress | ❌ |
| 23 | Ente.io | https://help.ente.io/self-hosting/ | ❌ |
| 24 | Thumbor | https://github.com/thumbor/thumbor | ❌ |
| 25 | Metube | https://github.com/alexta69/metube | ✅ |
| 26 | Appsmith | https://docs.appsmith.com/getting-started/setup/installation-guides/docker | ✅ |
| 27 | Authentik | https://docs.goauthentik.io/docs/installation/docker-compose | ❌ |
| 28 | Plausible Analytics | https://github.com/plausible/community-edition | 🔨 |
| 29 | Grocy | https://github.com/grocy/grocy | ✅ |
| 30 | Frappe | https://github.com/frappe/frappe | ✅ |
| 31 | NodeRed | https://nodered.org/docs/getting-started/docker | ✅ |
| 32 | Jellyfin | https://jellyfin.org/docs/general/installation/container | 🔨 |
| 33 | Emby | https://emby.media/docker-server.html | ❌ |
| 34 | Fider | https://fider.io/docs/hosting-instance | ❌ |
| 35 | Strapi | https://github.com/strapi/strapi | 🔨 |
| 36 | Hasura | https://hasura.io/docs/latest/getting-started/docker-simple/  |  ❌ |
| 37 | Metabase | https://www.metabase.com/docs/latest/installation-and-operation/running-metabase-on-docker | 🔨 |
| 38 | Wireguard VPN | https://github.com/wg-easy/wg-easy | ✅ |
| 39 | Chevereto | https://github.com/chevereto/docker | ✅ |
| 40 | Forgejo | https://forgejo.org/docs/latest/admin/installation-docker/ | 🔨 |
| 41 | Glance | https://github.com/glanceapp/glance?tab=readme-ov-file#docker | ❌ |
| 42 | Vaultwarden | https://github.com/dani-garcia/vaultwarden | ✅ |
| 43 | Uptime Kuma | https://github.com/louislam/uptime-kuma| ✅ |
| 44 | PostgreSQL Backup S3 | https://github.com/eeshugerman/postgres-backup-s3 | ✅ |
| 45 | MySQL / MariaDB Backup S3 | https://hub.docker.com/r/databack/mysql-backup | ✅ |
| 46 | MongoDB Backup S3 | https://github.com/swiftwave-org/app-store/tree/main/build/mongodb-backup | ✅ |
| 47 | Huly.io | https://github.com/hcengineering/platform | ❌ |
| 48 | Pocketbase | https://pocketbase.io/docs/going-to-production | ✅ |
| 49 | yt-dlp-web-gui | https://github.com/marcopeocchi/yt-dlp-web-ui/tree/master | ✅ |
| 50 | Cloudflare Tunnel | https://www.cloudflare.com/en-in/products/tunnel/ | ✅ |
| 51 | Dozzle | https://dozzle.dev/ | ✅ |
| 52 | Portainer | https://www.portainer.io/ | ✅ |
| 53 | Filestash | https://www.filestash.app/ | ✅ |
| 54 | Databag | https://github.com/balzack/databag | ✅ |
| 55 | Cockpit (CMS) | https://getcockpit.com/  | 🔨 |
| 56 | Minecraft | https://hub.docker.com/r/itzg/minecraft-server | ✅ |
| 57 | Kimai | https://hub.docker.com/r/kimai/kimai2 | 🔨 |
| 58 | Mongo Express | https://hub.docker.com/_/mongo-express | 🔨 |
| 59 | Collabora Online (server) | https://www.collaboraoffice.com/ | 🔨 |
| 60 | Cloud  Torrent | https://github.com/jpillora/cloud-torrent  | ✅ |
| 70 | Directus | https://docs.directus.io/self-hosted/docker-guide.html | 🔨 |
| 71 | EteSync | https://hub.docker.com/r/victorrds/etesync | 🔨 |
| 72 | Etherpad | https://hub.docker.com/r/elestio/etherpad | 🔨 |
| 73 | FusionAuth | https://fusionauth.io/docs/get-started/download-and-install/docker | 🔨 |
| 74 | Hedgedoc | https://hub.docker.com/r/linuxserver/hedgedoc |  🔨 |
| 75 | Imagor | https://github.com/cshum/imagor |  🔨 |
| 76 | Invoice Ninja | https://github.com/invoiceninja/invoiceninja | 🔨 |
| 77 | Maildev | https://github.com/maildev/maildev | 🔨 |
| 78 | Kutt | https://hub.docker.com/r/kutt/kutt | 🔨 |
| 79 | Mattermost | https://docs.mattermost.com/install/install-docker.html | 🔨 |
| 80 | OneDev | https://docs.onedev.io/installation-guide/run-as-docker-container | 🔨 |
| 81 | Plex | https://hub.docker.com/r/linuxserver/plex | 🔨 |
| 82 | Poste.io | https://poste.io/ | 🔨 |
| 83 | SurrealDB (in memory) | https://hub.docker.com/r/surrealdb/surrealdb | ✅ |
| 84 | Syncthing | https://syncthing.net/ | 🔨 |
| 85 | Typebot | https://docs.typebot.io/self-hosting/deploy/docker | 🔨 |
| 86 | Wiki.js | https://docs.requarks.io/install/docker | 🔨 |
| 87 | Azuriom | https://azuriom.com/en/ | ❌ |
| 88 | Pelican | https://pelican.dev/ | ❌ |
| 89 | Paperless NGX | https://github.com/paperless-ngx/paperless-ngx | ❌ |
| 90 | Pterodactyl | https://pterodactyl.io/ | ❌ |
| 91 | Penpot | https://penpot.app/ | ✅ |
| 92 | EdgeDB | https://www.edgedb.com/ | ✅ |
| 93 | Directus | https://docs.directus.io/self-hosted/quickstart.html | ❌ |
| 94 | Zitadel | https://zitadel.com/docs/self-hosting/deploy/compose | ❌ |
| 95 | Zammad | https://github.com/zammad/zammad-docker-compose | ❌ |
| 96 | n8n | https://github.com/n8n-io/n8n-hosting/blob/main/docker-compose/withPostgres/docker-compose.yml | ✅ |

**Notes -**
- _pihole_ can't be onboarded due to conflict with systemd-resolve and local dns server. Disabling this on a system, can cause issue on other stuffs. Required many manual tasks to do those correctly. Also most of the homerouter dont allow to set dns server on custom port. So only way to disable system-resolved andd setup ingress for port 53. Hence pihole will not be onboarded in  app-store.

**Comment by tanmoysrt** [comment]
This has a good list of applications which can be self-hosted
https://awesome-selfhosted.net/

**Comment by tonislcs** [comment]
This could be cool https://github.com/chevereto/docker also has arm64 support

**Reply by tanmoysrt** [reply]
I haven't used cheverato personally before.
Should, we go for `chevereto-free` or `chevereto`, as I can access very few features in `chevereto` free version ?

**Reply by tanmoysrt** [reply]
`chevereto` added to app store

**Comment by znowfox** [comment]
i suggest changing gitea for forgejo (or adding it)

**Reply by tanmoysrt** [reply]
We can add `forgejo` also. Manyone prefer `gitea` for simple git hosting.

**Reply by Nucceteere** [reply]
Will Forgejo be added? It seems that it's still not added.

**Reply by tanmoysrt** [reply]
Hi @Nucceteere 

I am currently not adding up new apps to app store. But, anyone is free to add new apps in [store](https://github.com/swiftwave-org/app-store).

I am bit focusing on architecture restructure as of now > https://github.com/orgs/swiftwave-org/discussions/568#discussioncomment-12208001

**Reply by Nucceteere** [reply]
> I am currently not adding up new apps to app store. But, anyone is free to add new apps in [store](https://github.com/swiftwave-org/app-store).
> 
> I am bit focusing or architecture restructure as of now

Thanks for the reply @tanmoysrt

I will look into making pull requests tonight.

**Reply by lucastucious** [reply]
Following this topic

**Comment by tanmoysrt** [comment]
Any suggestion for database backup tools for PostgreSQL / MariaDB / MySQL / MongoDB. which may support to backup to s3 or any other storage at frequent interval.

as well as database restore tools (if available any)

**Reply by znowfox** [reply]
https://github.com/eeshugerman/postgres-backup-s3

**Comment by r1ude** [comment]
I would love to see Paperless-NGX and Pterodactyl in the AppStore.

**Reply by tanmoysrt** [reply]
This two ?
- https://github.com/paperless-ngx/paperless-ngx
- https://pterodactyl.io/

**Reply by r1ude** [reply]
> This two ?
> 
> * https://github.com/paperless-ngx/paperless-ngx
> * https://pterodactyl.io/

Yes these two :D

**Comment by tonislcs** [comment]
If i was you i wouldn't add pterodactyl at the moment since their team switched to their new project https://pelican.dev/ and their main deploy will be done via docker image so it will be easy to add

**Reply by tanmoysrt** [reply]
so pelican is kind of revamped version of pterodactyl ?

**Reply by r1ude** [reply]
Its a OOS Team that split up, like ownCloud and Nextcloud.

**Reply by tonislcs** [reply]
Yeah i think cause the maintainer of pterodactyl wouldn't accept prs don't quote me on that tho

**Reply by tanmoysrt** [reply]
Let's add both in pending list, at the time of onboarding , we can onboard both also. Now, it depends on the user, he wants to install which one.

**Comment by tonislcs** [comment]
Would be ncie to add https://azuriom.com/en/

**Reply by tanmoysrt** [reply]
Thanks will add that.

**Reply by tonislcs** [reply]
Do you know when you will do so i know if to self-host it normally through nginx etc or use swiftwave Thanks :)

**Reply by tanmoysrt** [reply]
We can build custom docker image for that. Installation steps are already in docs, so no problem

**Comment by tanmoysrt** [comment]
Lot of apps are tested to onboard (will update the readme with that). Thanks @znowfox for preparing and testing lot of apps. But before next onboarding, revamping of application grouping is on high priority - https://github.com/swiftwave-org/swiftwave/issues/883

After onboarded supabase(it has around 12 services), realised that, it's tough to change configuration of a app (we need to go inside each service to do some tweaks). Currently, there is no option to delete/pause/resume a application group together. Theses issues need to  be fixed before next onboarding.

edit: all the tested apps added in the discussion post. will be onboarded once the application grouping improved

**Comment by zaniar** [comment]
How about copy the list from other similar projects?

Coolify https://github.com/coollabsio/coolify/tree/main/templates/compose  
Kubero https://github.com/kubero-dev/kubero/tree/main/services  

I'll put my quick test to https://gist.github.com/zaniar/fa49767eb9a9333da71ab945bc6d3754
I'll make some PR later after I get the hang of it. Meanwhile, feel free to copy them to the main repo if you think they are good enough to be included.

**Reply by tanmoysrt** [reply]
Thanks @zaniar for sharing the links. I have also checked available paas's onboarded apps list, but as onboarding apps are time consuming, that's why adding apps as per priority and requirement of users.

**Reply by zaniar** [reply]
Is there any guide to create One-Click App? I would like to help create some.
Is there a way to run my own App Store? So I can test the apps I made before merged back to the official App Store.

**Reply by tanmoysrt** [reply]
I haven't write the docs speicific to the format, but you can check the existing templates to figure out - https://github.com/swiftwave-org/app-store

**store.json** is the listing file and in **stacks** folder you will find specific endpoints.

Some examples -
- https://github.com/swiftwave-org/app-store/blob/main/stacks/wordpress-mariadb.yaml
- If you need access to docker socket, https://github.com/swiftwave-org/app-store/blob/main/stacks/dozzle.yaml
- If you need to expose tcp port for your app, https://github.com/swiftwave-org/app-store/blob/main/stacks/redis.yaml

---

For development, you can't directly modify the app store url. But you can test out your own app template by passsing the url of your yaml in the url - 

`https://<swiftwave endpoint>/dashboard/deploy/app-store/install?stack=<yaml_file_url>`

For example - 
`https://example.com:3333/dashboard/deploy/app-store/install?stack=https://raw.githubusercontent.com/swiftwave-org/app-store/main/stacks/chevereto-mariadb.yaml`

**Reply by zaniar** [reply]
I'll put my quick test to https://gist.github.com/zaniar/fa49767eb9a9333da71ab945bc6d3754
I'll make some PR later after I get the hang of it. Meanwhile, feel free to copy them to the main repo if you think they are good enough to be included.

I also updated my original comment so people can find the link to my gist easier.

**Comment by ardilabcode** [comment]
Add Penpot (https://penpot.app) and Odoo (https://www.odoo.com) please. I think many people will need it, like me.

**Reply by tanmoysrt** [reply]
Will add penpot this weekend. But for odoo, I am not sure. Looks like, It has a different kind of OSS license. Need to check before adding. Anyhow you can deploy manually that.

**Reply by ardilabcode** [reply]
Wow, lot of thanks... great news you bring Penpot on Swiftwave :)

Can you please provide some official tutorial on deploying Odoo ?

**Reply by tanmoysrt** [reply]
This can help you > https://www.digitalocean.com/community/tutorials/how-to-install-odoo-on-ubuntu-20-04-with-docker

**Reply by tanmoysrt** [reply]
Btw, If you are looking for open source ERP and want to self-host, you can try-out  ERPNext (https://github.com/frappe/erpnext).
In App Store, frappe app has been already added, so it's easy to install ERPNext there.

**Reply by ardilabcode** [reply]
i appreciate for your prompt reply :)

**Reply by tanmoysrt** [reply]
Hi @ardilabcode 

Penpot has been added. 2 variants added.
- With SMTP Configuration
- Without SMTP Configuration.

Try out. If you face any issue, post here or [repo's](https://github.com/swiftwave-org/app-store) issue tracker.

**Reply by ardilabcode** [reply]
Wow, awesome. I have successfully installed penpot on server with Swiftwave.. Thank you so much 👍👍👍👍👍. Setting SMTP with AWS SES works like a charm.

**Comment by subrotosanyal** [comment]
@tanmoysrt 
Requesting to add [n8n](https://n8n.io/)
Adding it to the Swiftwave App Store will further empower users with workflow automation capabilities that will easily connect and automate tasks between different apps and services. Having open-source flexibility with visual workflow builders, n8n can be used to help users of Swiftwave streamline operations, save time, and thereby increase productivity even without requiring in-depth coding knowledge.

**Reply by that0n3guy** [reply]
I'd also like to add that the n8n community and influencer world is blowing up.   If you were to reach out to 5 of main 10 that are producing n8n content, you could probably get some good/free mentions of swiftwave.

**Reply by tanmoysrt** [reply]
Looks like a simple configuration - https://github.com/n8n-io/n8n-hosting/blob/main/docker-compose/withPostgres/docker-compose.yml

**Reply by tanmoysrt** [reply]
@subrotosanyal @that0n3guy 

![image](https://github.com/user-attachments/assets/68cc815e-a135-4f9f-b408-e2bdd47f3caa)

Added n8n. Try out.
If you face issue lmk.

**Reply by that0n3guy** [reply]
You and I were working on it at the same time I think (I sent a PR: https://github.com/swiftwave-org/app-store/pull/22, mine uses external DB).    I tested yours, it works.   It appears that n8n requires https though so might want to put that in the details...  or just let the user figure it out. (it will give you an error on http).

**Comment by NRisticHtec** [comment]
Add Navidrome and Huginn app please.

Here is Huginn working .env: 
```
# timezone only allow ruby aliases 
# https://gist.github.com/eikes/5a64b661022c756bd6522ed94770e2a6 Europe/Paris not working
# https://gist.github.com/lucascaton/bec400d18f7dcda61275
TIMEZONE=Paris
DELAYED_JOB_MAX_RUNTIME=10 # allow running jobs for 10 minutes
  
HUGINN_DATABASE_USERNAME=huginn
HUGINN_DATABASE_PASSWORD=mysecretpassword
HUGINN_DATABASE_ADAPTER=postgresql

EMAIL_FROM_ADDRESS=huginn@mailgun-userid.mailgun.org
SMTP_DOMAIN=mailgun-userid.mailgun.org
SMTP_USER_NAME=postmaster@mailgun-userid.mailgun.org
SMTP_PASSWORD=mailgun-smtp-password
SMTP_SERVER=smtp.mailgun.org
SMTP_PORT=587
SMTP_AUTHENTICATION=plain
SMTP_ENABLE_STARTTLS_AUTO=true
SEND_EMAIL_IN_DEVELOPMENT=true
```

**Comment by tanmoysrt** [comment]
We are doing some rewrite of swiftwave architecture.
For the time being, we are not adding up apps in store actively.
Anyway, You might raise a PR in app store, we can review and add that.

**Comment by chillpilllike** [comment]
Add odoo, very much needed

Odoo.com
